plugins/linuxv2/l3plugin/l3plugin.go

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2018 Cisco and/or its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:generate protoc --proto_path=../model/l3 --proto_path=${GOPATH}/src --gogo_out=../model/l3 l3.proto
+//go:generate adapter-generator --descriptor-name ARP --is-proto --value-type *l3.LinuxStaticARPEntry --from-datasync --import "../model/l3" --output-dir "descriptor"
+//go:generate adapter-generator --descriptor-name Route --is-proto --value-type *l3.LinuxStaticRoute --from-datasync --import "../model/l3" --output-dir "descriptor"
+
+package l3plugin
+
+import (
+	"github.com/ligato/cn-infra/infra"
+	scheduler "github.com/ligato/cn-infra/kvscheduler/api"
+	"github.com/ligato/cn-infra/logging/measure"
+
+	"github.com/ligato/vpp-agent/plugins/linuxv2/ifplugin"
+	"github.com/ligato/vpp-agent/plugins/linuxv2/l3plugin/descriptor"
+	"github.com/ligato/vpp-agent/plugins/linuxv2/l3plugin/descriptor/adapter"
+	"github.com/ligato/vpp-agent/plugins/linuxv2/l3plugin/linuxcalls"
+	"github.com/ligato/vpp-agent/plugins/linuxv2/nsplugin"
+)
+
+// L3Plugin configures Linux routes and ARP entries using Netlink API.
+type L3Plugin struct {
+	Deps
+
+	// From configuration file
+	disabled  bool
+	stopwatch *measure.Stopwatch
+
+	// system handlers
+	l3Handler linuxcalls.NetlinkAPI
+
+	// descriptors
+	arpDescriptor   *descriptor.ARPDescriptor
+	routeDescriptor *descriptor.RouteDescriptor
+}
+
+// Deps lists dependencies of the interface plugin.
+type Deps struct {
+	infra.PluginDeps
+	Scheduler scheduler.KVScheduler
+	NsPlugin  nsplugin.API
+	IfPlugin  ifplugin.API
+}
+
+// Config holds the nsplugin configuration.
+type Config struct {
+	Stopwatch bool `json:"stopwatch"`
+	Disabled  bool `json:"disabled"`
+}
+
+// Init initializes and registers descriptors for Linux ARPs and Routes.
+func (plugin *L3Plugin) Init() error {
```

Comment:
`plugin` -> `p`

Let's keep all method receivers short, ideally just a single character. It's generally the most repeated variable for methods and such variables should not be so verbose.